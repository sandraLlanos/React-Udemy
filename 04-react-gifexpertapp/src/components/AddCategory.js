import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
        console.log('handleInputChange is called');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('HandldSubmit is called');
        
        if ( inputValue.trim().length > 2 ) {
            console.log('setCategories is called');
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <p> {inputValue} </p>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
