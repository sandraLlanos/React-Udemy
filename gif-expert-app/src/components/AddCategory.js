import React, { useState } from 'react'

const AddCategory = () => {
    const [inputValue, setInputValue] = useState('hello word!');
    const handleInputValue = (e) => {
        setInputValue(e.target.value);
    } 
    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('Form submited');
    }
    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputValue}
            />
        </form>
    )
}

export default AddCategory
