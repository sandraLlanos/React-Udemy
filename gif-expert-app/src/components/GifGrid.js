import React from 'react'

export const GifGrid = ( {category } ) => {

    const getGifs = async() => {
        const apiUrl = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=XRnsOEVygIHs5bH4hMpOall24oypnnmH';

        const resp = await fetch( apiUrl );
        const {data} = await resp.json();
        // console.log(data[0].id)
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    getGifs();

    return (
        <>
            <h3> {category} </h3>
            
        </>
    )
}
