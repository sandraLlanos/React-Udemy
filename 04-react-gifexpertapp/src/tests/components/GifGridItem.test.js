import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('test in <GifGridItem />', () => {

    const title = 'a title';
    const url = 'https://localhost/something.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('should show GifGridItem component successfully ', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should be have a paragraph with a title', () => {
        const p = wrapper.find('p');
        console.log(p);
        expect(p.text().trim()).toBe(title);
    });
    test('should be have the image equals to url and alt', () => {
        const img = wrapper.find('img');
        console.log(img.html());
        console.log(img.props());
        console.log(img.props().src);
        console.log(img.prop('alt'));
        expect(img.props().src).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
    test('should be have animate__fadeIn', () => {
        const div = wrapper.find('div');
        console.log(div.props().className);
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    });
});