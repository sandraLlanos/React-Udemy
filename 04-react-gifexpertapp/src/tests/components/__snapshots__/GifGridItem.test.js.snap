// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifGridItem /> should show GifGridItem component successfully  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="a title"
    src="https://localhost/something.jpg"
  />
  <p>
     
    a title
     
  </p>
</div>
`;
